tests/integration/test_hal_exporter.py

```diff
@@ -20,24 +20,22 @@
 # granted to it by virtue of its status as an Intergovernmental Organization
 # or submit itself to any jurisdiction.
 
-"""Integration tests for the TEI exporter."""
-
 from __future__ import absolute_import, division, print_function
 
 import os
 import pkg_resources
 
-import pytest
+from lxml import etree
 
-from inspirehep.modules.hal import tei
+from inspirehep.modules.hal.api import convert_to_tei
 from inspirehep.utils.record_getter import get_db_record
 
 
 def test_format_tei(app):
-    expected = pkg_resources.resource_string(
-        __name__, os.path.join('fixtures', 'test_tei_record.xml'))
-
     record = get_db_record('lit', 1407506)
-    result = tei.tei_response(record)
 
-    assert result.strip() == expected.strip()
+    schema = etree.XMLSchema(etree.parse(pkg_resources.resource_stream(
+        __name__, os.path.join('fixtures', 'aofr-sword.xsd'))))
+    result = etree.fromstring(convert_to_tei(record).encode('utf8'))
```

Comment:
These two lines look very scary, but it's a much better test than trying to match an XML file byte by byte (which is pointless).